packages/components/src/components/Alert/Alert.test.tsx

```diff
@@ -18,292 +18,326 @@ jest.mock('react-native/Libraries/Utilities/useColorScheme', () => {
-        expect(component.queryByText('Description text')).toBeTruthy()
-        expect(component.queryByText('Sample children content')).toBeTruthy()
-        expect(component.queryByText('Primary test button')).toBeTruthy()
-        expect(component.queryByText('Secondary test button')).toBeTruthy()
+        render(<Alert variant="error" expandable {...commonProps} />)
+
+        fireEvent.press(screen.getByRole('tab'))
+        expect(screen.getByText('Header text')).toBeOnTheScreen()
+        expect(screen.getByText('Description text')).toBeOnTheScreen()
+        expect(screen.getByText('Sample children content')).toBeOnTheScreen()
+        expect(screen.getByText('Primary test button')).toBeOnTheScreen()
+        expect(screen.getByText('Secondary test button')).toBeOnTheScreen()
       })
     })
 
     describe('Initialize expanded', () => {
-      beforeEach(() => {
-        component = render(
+      it('should render everything when initializeExpanded is set to true', () => {
+        render(
           <Alert
             variant="error"
-            header={headerText}
-            description={descriptionText}
             expandable
             initializeExpanded={true}
-            primaryButton={primaryButtonProps}
-            secondaryButton={secondaryButtonProps}>
-            {children}
-          </Alert>,
+            {...commonProps}
+          />,
         )
       })
 
-      it('should render everything when initializeExpanded is set to true', () => {
-        expect(component.queryByText('Header text')).toBeTruthy()
-        expect(component.queryByText('Description text')).toBeTruthy()
-        expect(component.queryByText('Sample children content')).toBeTruthy()
-        expect(component.queryByText('Primary test button')).toBeTruthy()
-        expect(component.queryByText('Secondary test button')).toBeTruthy()
-      })
+      expect(screen.getByText('Header text')).toBeOnTheScreen()
+      expect(screen.getByText('Description text')).toBeOnTheScreen()
+      expect(screen.getByText('Sample children content')).toBeOnTheScreen()
+      expect(screen.getByText('Primary test button')).toBeOnTheScreen()
+      expect(screen.getByText('Secondary test button')).toBeOnTheScreen()
+    })
 
-      it('should render only the header after being collapsed', () => {
-        fireEvent.press(component.getByRole('tab'))
-        expect(component.queryByText('Header text')).toBeTruthy()
-        expect(component.queryByText('Description text')).toBeNull()
-        expect(component.queryByText('Sample children content')).toBeNull()
-        expect(component.queryByText('Primary test button')).toBeNull()
-        expect(component.queryByText('Secondary test button')).toBeNull()
-      })
+    it('should render only the header after being collapsed', () => {
+      render(
+        <Alert
+          variant="error"
+          expandable
+          initializeExpanded={true}
+          {...commonProps}
+        />,
+      )
```

Comment:
Did this test need to render everything again or would it still be on screen in collapsed form?  Curious since the render for the prior test is outside the test and doesn't change anything. Maybe this is also bumping up against the linter though.

Reply:
Yes, the test would fail with `'render' method has not been called` if there hasn't been a "fresh" render called prior to the test. I suspect the previous test didn't fail since render is called right before, even if it wasn't within the it(). Moved the render to be within.